Solutions/Azure Firewall/Analytic Rules/Azure Firewall - Multiple Sources Affected by the Same TI Destination.yaml

```diff
@@ -5,40 +5,45 @@ description: |
   
   Configurable Parameters:
   
-  - Minimum affected threshold - alert only if more than this number of hosts affected. Default is set to 5.'
+  - Minimum affected threshold - alert only if more than this number of hosts affected. Default is set to 5.
+  - Recommendation is to use the new resource specific logs. If you are using both, the TiTraffic Count will be duplicated.'
 severity: Medium
 status: Available
 requiredDataConnectors:
   - connectorId: AzureFirewall
     dataTypes: 
       - AzureDiagnostics
+      - AZFWThreatIntel
 queryFrequency: 1h
-queryPeriod: 1h
+queryPeriod: 1d
 triggerOperator: gt
 triggerThreshold: 1
 tactics:
   - Exfiltration
   - CommandAndControl
 relevantTechniques: []
 query: |
-  let RunTime = 1h; 
+  let RunTime = 1d; 
   let StartRunTime = 1d; 
   let EndRunTime = StartRunTime - RunTime; 
   let MinAffectedThershold = 5; 
-  AzureDiagnostics 
-  | where TimeGenerated  between (ago(StartRunTime) .. ago(EndRunTime)) 
-  | parse msg_s with * "from " srcip ":" srcport " to " dsturl ":" dstport  "." * "ThreatIntel: " ThreatIntel 
-  | where OperationName == "AzureFirewallThreatIntelLog" | summarize TiTrafficCount = count(), dCountSourceIps = dcount(srcip), AffectedIps = make_set(srcip) by dsturl, ThreatIntel, srcip 
-  | where dCountSourceIps > MinAffectedThershold | order by TiTrafficCount desc
-  | extend IPCustomEntity = srcip, URLCustomEntity = dsturl
+  union isfuzzy=true
+  (AzureDiagnostics 
+  | where TimeGenerated  between (ago(StartRunTime) .. ago(EndRunTime))
+  | where OperationName == "AzureFirewallThreatIntelLog"
+  | parse msg_s with * "from " SourceIp ":" SourcePort:int " to " Fqdn ":" DestinationPort:int  "." * "Action: Deny. " ThreatDescription),
+  (AZFWThreatIntel
+  | where TimeGenerated between (ago(StartRunTime) .. ago(EndRunTime)))
+  | summarize TiTrafficCount = count(), dCountSourceIps = dcount(SourceIp), AffectedIps = make_set(SourceIp) by Fqdn, ThreatDescription 
```

Comment:
You can add a high number like 1000 or 10000. 
Practically speaking, it will be impossible for the SOC Analyst to event check more than a few hundred IPs or investigate them. Nevertheless, we can have a high number to ensure both completeness and performance or the query.

Reply:
The query has been updated accordingly.